crates/ruff_python_parser/src/string.rs

```diff
@@ -178,6 +178,12 @@ impl<'a> StringParser<'a> {
                     'N' if !self.kind.is_any_bytes() => self.parse_unicode_name()?,
                     // Special cases where the escape sequence is not a single character
                     '\n' => return Ok(String::new()),
+                    '\r' => {
+                        if self.peek() == Some('\n') {
+                            self.next_char();
+                        }
+                        return Ok(String::new());
```

Comment:
What about \r with no \n after it? That’s also a valid newline pattern.

Reply:
It's also covered here. We'll match on \r and peek to see if it's followed by \n and if so we'll skip that character.